q2_composition/tests/test_diff_abundance_plot.py

```diff
@@ -0,0 +1,293 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import tempfile
+from pathlib import Path
+
+from qiime2.plugin.testing import TestPluginBase
+from qiime2 import Artifact
+
+from q2_composition import (da_barplot, DataLoafPackageDirFmt)
+
+
+class TestBase(TestPluginBase):
+    package = 'q2_composition.tests'
+
+    def setUp(self):
+        super().setUp()
+        self.dataloaf1 = Artifact.load(self.get_data_path('dataloaf.qza'))
+        self.dataloaf2 = \
+            Artifact.load(self.get_data_path('dataloaf-hmp1-body-habitat.qza'))
+
+
+class TestDABarplot(TestBase):
+
+    def test_basic_dl1(self):
+        with tempfile.TemporaryDirectory() as output_dir:
+            output_dir = Path(output_dir)
+            _ = da_barplot(output_dir,
+                           self.dataloaf1.view(DataLoafPackageDirFmt),
+                           feature_id_label='taxon')
+            self.assertTrue((output_dir / 'index.html').exists())
+
+            tongue_path = output_dir / 'bodysitetongue-ancombc-barplot.html'
+            self.assertTrue(tongue_path.exists())
+
+            right_palm_path = \
+                output_dir / 'bodysiteright-palm-ancombc-barplot.html'
+            self.assertTrue(right_palm_path.exists())
+
+            left_palm_path = \
+                output_dir / 'bodysiteleft-palm-ancombc-barplot.html'
+            self.assertTrue(left_palm_path.exists())
+
+            # file shouldn't exist for reference column
+            gut_path = \
+                output_dir / 'bodysitegut-ancombc-barplot.html'
+            self.assertFalse(gut_path.exists())
+
+            # spot check for a feature id
+            self.assertTrue(
+                'd29fe3c70564fc0f69f2c03e0d1e5561' in
+                open(tongue_path).read())
+            self.assertTrue(
+                'd29fe3c70564fc0f69f2c03e0d1e5561' in
+                open(right_palm_path).read())
+            self.assertTrue(
+                'd29fe3c70564fc0f69f2c03e0d1e5561' in
+                open(left_palm_path).read())
+
+    def test_basic_dl2(self):
+        with tempfile.TemporaryDirectory() as output_dir:
+            output_dir = Path(output_dir)
+            _ = da_barplot(output_dir,
+                           self.dataloaf2.view(DataLoafPackageDirFmt))
+            self.assertTrue((output_dir / 'index.html').exists())
+
+            feces_path =  \
+                output_dir / 'body_habitatUBERON:feces-ancombc-barplot.html'
```

Comment:
An example of the escape issue, this filepath is illegal on NTFS, so Windows machines can't unzip these visualizations without the tool renaming something.